pkg/upgrade/control_plane.go

```diff
@@ -594,6 +596,9 @@ func upgradeSuffix(upgradeID string) string {
 	return ".upgrade." + upgradeID
```

Comment:
Can this break `validation.IsDNS1123Subdomain`? For example, what happens if my upgrade id is `.123` which would make this `.upgrade..123`. Is that valid?

Reply:
Good question! `.123` won't pass that validation, and therefore is not a valid upgrade id
```
go run ./main.go --cluster-namespace cns --cluster-name cn --kubernetes-version v1.15.3 --upgrade-id .123
Error: upgrade ID: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
upgrade ID: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
github.com/vmware/cluster-api-upgrade-tool/pkg/upgrade.NewControlPlaneUpgrader
        /Users/andy/code/capi-dev/cluster-api-upgrade-tool/pkg/upgrade/control_plane.go:92
main.main.func1
        /Users/andy/code/capi-dev/cluster-api-upgrade-tool/main.go:51
```

Reply:
oh, wonderful